infix/token.go

```diff
@@ -0,0 +1,35 @@
+package infix 
+
+type (
+	Token int
+)
+
+const (
+	Illegal Token = iota
+	Number
+	LParen
+	RParen
+	OR
+	AND
+	XOR
+	NOT
+	SHL
+	SHR
+	EOF
+)
+
+func (t Token) String() string {
+	switch t {
+	case LParen: return "("
+	case RParen: return ")"
+	case OR: return "|"
+	case AND: return "&"
+	case XOR: return "^"
+	case NOT: return "~"
+	case SHL: return "<<"
+	case SHR: return ">>"
+	case Illegal: return "<ileggal>"
+	case EOF: return "EOF"
+	}
+	panic("invalid token")
```

Comment:
This makes me think on the need of the language to have some sort of enumeration that does not allow the type to exists with a invalid value (validated statically). For these cases the simplification of only having consts feels flat =/